modules/afsocket/afinet-dest.c

```diff
@@ -144,7 +144,7 @@ afinet_dd_verify_callback(gint ok, X509_STORE_CTX *ctx, gpointer user_data)
   if (ok && current_cert == cert && self->hostname
       && (tls_context_get_verify_mode(self->tls_context) & TVM_TRUSTED))
     {
-      ok = tls_verify_certificate_name(cert, self->hostname);
+      ok = tls_context_ignore_hostname_check(self->tls_context) || tls_verify_certificate_name(cert, self->hostname);
```

Comment:
It is a mistake to start adding tls_context_ignore_hostname_check() to all call sites where we want to avoid the hostname checking.

Also, the flag itself is in the TLSContext layer and we implement the checking of it in the TLSVerifier callback. :/

The TLSVerifier is not the best interface, I might add, so not necessarily your fault.

Maybe if we added the TLSContext argument to tls_verify_certificate_name() then the checking of the flag could go inside that function?

That's still not very nice, but would be better than having to replicate this check in every call site.

Reply:
Done